include/oboe/AudioStreamBase.h

```diff
@@ -195,6 +195,28 @@ class AudioStreamBase {
     bool                            mFormatConversionAllowed = false;
     // Control whether and how Oboe can convert sample rates to achieve optimal results.
     SampleRateConversionQuality     mSampleRateConversionQuality = SampleRateConversionQuality::None;
+
+    /** Validate the existing stream configurations */
+    virtual Result isValidConfig() {
+        if (mFormat < AudioFormat::Invalid || mFormat > AudioFormat::Float) {
+            return Result::ErrorInvalidFormat;
+        }
+        if ((mSharingMode != SharingMode::Exclusive && mSharingMode != SharingMode::Shared) ||
+            (mDirection != Direction::Input && mDirection != Direction::Output) ||
+            (mPerformanceMode != PerformanceMode::None && mPerformanceMode != PerformanceMode::PowerSaving &&
+             mPerformanceMode != PerformanceMode::LowLatency)    ||
+            (mUsage < Usage::Media || mUsage > Usage::Assistant) ||
+            (mContentType < ContentType::Speech || mContentType > ContentType::Sonification) ||
```

Comment:
No comparison operators for enums.
The goal was to have more informative error codes. So for these I would just let AAudio catch them.

Reply:
updated, please take a look again